src/model/page-config-resolver.js

```diff
@@ -283,13 +283,180 @@ class JsonLdParser {
   }
 }
 
+/**
+ * Class the describes the microdata found in the page
+ */
+class Microdata {
+  /**
+   * @param {!Array<!{id: number, properties: Object<string, (string|boolean)>}>} entries
+   */
+  constructor(entries) {
+    this.entries = entries;
+  }
+
+  /**
+   * Returns the first page configuration found, if present
+  * @return {?PageConfig} pageConfig found
+  */
+  getPageConfig() {
+    const pageConfigs = [];
+    let locked = false;
+    let productId = null;
+    this.entries.forEach(item => {
+      Object.keys(item.properties).forEach(property => {
+        if (property == 'isAccessibleForFree') {
+          locked = !item.properties[property];
+        }
+        if (property == 'productID') {
+          productId = item.properties[property];
+        }
+      });
+      if (productId != null) {
+        pageConfigs.push(new PageConfig(productId, locked));
+      }
+    });
+    if (pageConfigs.length == 0) {
+      return null;
+    } else {
+      return pageConfigs[0];
+    }
+  }
+}
+
 class MicrodataParser {
   /**
    * @param {!Doc} doc
    */
   constructor(doc) {
     /** @private @const {!Doc} */
     this.doc_ = doc;
+    /** @private {?boolean} */
+    this.access_ = null;
+    /** @private {?string} */
+    this.productID_ = null;
```

Comment:
`productId_`. I know that Schema.org has a different styleguide, but JS styleguide in JS code wins.